students/A_Zach/lesson03/mailroom1.py

```diff
@@ -0,0 +1,161 @@
+import sys
+
+#Functions
+#Add a donation to a person on the list
+def new_donation(inp,n,d):
+    response_donation = input("Please enter an amount to donate >")
+    quit_prog(response_donation)
+    
+    i = 0
+    
+    #iterate through the database until the requested name is found
+    while i < len(n):
+        #Add the donation to the name provided
+        if n[i] == inp.title():
+            d[i].append(float(response_donation))        
+            break
+        #Increase iteration if name has not been reached
+        i += 1
+    return
+
+#Show list of names
+def show_list(inp,n):
+    #print names of donors
+    for i in n:
+        print(i)
+    return
+    
+#Add a new name to the list
+def search_name(inp,n,d):
+    for i in n:
+        #if the name is in the list add a new donation. If not add the name and a new donation.
+        if inp.title() == i:
+            new_donation(inp,n,d)
+            #leave loop if name has been found and donation added
+            return
+    n.append(inp.title())
+    d.append([])
+    new_donation(inp,n,d)
+    return
+
+#make the program quit
+def quit_prog(p):
+    if p.lower() == 'quit':
+        print("Thank you. Goodbye!")
+        sys.exit()
+
+#generate an email1
+def email(inp,n,d):
+    #find the index of n in the database
+    i = n.index(inp.title())
```

Comment:
you should be able to pass a full donor item here instead of searching the list again.